UnityProject/Assets/Scripts/Items/Weapons/AtmosGrenade.cs

```diff
@@ -0,0 +1,63 @@
+using Mirror;
+using Objects.Atmospherics;
+using UnityEngine;
+
+namespace Items.Weapons
+{
+	[RequireComponent(typeof(GasContainer))]
+	[RequireComponent(typeof(Grenade))]
+	public class AtmosGrenade : NetworkBehaviour, IServerInventoryMove, IExaminable
+	{
+		[Tooltip("SpriteHandler used for gas color overlay")]
+		public SpriteHandler gasIndicatorSpriteHandler;
+	
+		private GasContainer gasContainer;
+		private Grenade grenade;
+		
+		private readonly Color noGasColor = new(0,0,0,0);
+
+		public void Start()
+		{
+			gasContainer = GetComponent<GasContainer>();
+			grenade = GetComponent<Grenade>();
+			UpdateOverlay();
+			grenade.OnExpload.AddListener(ReleaseGas);
+		}
+
+		private void OnDestroy()
+		{
+			grenade.OnExpload.RemoveListener(ReleaseGas	);
+		}
+		
+		public void OnInventoryMoveServer(InventoryMove info)
```

Comment:
Shouldn't need to hook into the inventory movement code, Probably better to hook into the gas containers if there is any way of checking if it changes?

Reply:
humm doesn't look like there's an event for a gas mix changing, could add one but idk if it's worth it, 
any reviewers reading this I think it's fine to merge

Reply:
I could add an action to GasContainer and invoke it within GasMixLocal's setter or something? I believe that would work